main/auth/auth.py

```diff
@@ -132,6 +132,24 @@ def decorated_function(*args, **kwargs):
   return decorated_function
 
 
+def cron_required(f):
+  decorator_order_guard(f, 'auth.cron_required')
+
+  @functools.wraps(f)
+  def decorated_function(*args, **kwargs):
+    if config.PRODUCTION and 'X-Appengine-Cron' in flask.request.headers:
+      return f(*args, **kwargs)
+    if is_logged_in() and current_user_db().admin:
+      return f(*args, **kwargs)
+    if not is_logged_in() and flask.request.path.startswith('/api/'):
+      return flask.abort(401)
+    if not is_logged_in():
+      return flask.redirect(flask.url_for('signin', next=flask.request.url))
+    return flask.abort(403)
+
+  return decorated_function
+
+
```

Comment:
Perhaps a small difference, using:

```
def cron_required(f):
  decorator_order_guard(f, 'auth.cron_required')

  @functools.wraps(f)
  def decorated_function(*args, **kwargs):
    if config.PRODUCTION:
      if 'X-Appengine-Cron' in flask.request.headers:
        # TODO: check IP == 0.1.0.1
        return f(*args, **kwargs)
    else:
      if is_logged_in() and current_user_db().admin:
        return f(*args, **kwargs)
      if not is_logged_in():
        return flask.redirect(flask.url_for('signin', next=flask.request.url))
    return flask.abort(403)

  return decorated_function
```

To only allow admin access in non-PRODUCTION environment; also note that I removed the `/api/` check, as in my world that shouldn't be a cron end-point.